gammapy/datasets/make.py

```diff
@@ -566,11 +611,16 @@ def make_test_eventlist(observation_table,
     # positive defined, so it is necessary to translate the (0, 1)
     # interval of the random variable to (emax, e_min) in order to
     # have a decreasing power-law
-    index = spectral_index + 1
     e_min = Quantity(0.1, 'TeV')
     e_max = Quantity(100., 'TeV')
-    energy = Quantity((e_min.value - e_max.value) *
-                      random_state.power(a=index, size=n_events) + e_max.value, 'TeV')
+    energy = Quantity(sample_powerlaw(e_min.value, e_max.value,
+                                      spectral_index + 1, # TODO: need to add 1
+                                                          # to the index because
+                                                          # sample_powerlaw
+                                                          # subtracts it, why??!!!
+                                      size=n_events,
+                                      random_state=random_state),
+                      'TeV')
```

Comment:
@cdeil 
I found a bug: when sampling the energies for the simulated dataset, the spectrum was not a power-law (maybe `numpy.random.RandomState.power` is buggy, or it doesn't work as I thought).

I'm now using `sample_powerlaw`, which seems to work fine, BUT, I have to pass the index as `spectral_index + 1` because the function internally does `exp = 1. - gamma`:
https://github.com/gammapy/gammapy/blob/master/gammapy/utils/random.py#L127
I don't know why does the function need to subtract 1 to the specified index: bug or feature?